packages/run-protocol/src/contractSupport.js

```diff
@@ -1,8 +1,15 @@
 // @ts-check
 import { AmountMath } from '@agoric/ertp';
+import { M } from '@agoric/store';
 
 const { details: X } = assert;
 
+export const amountPattern = { brand: M.remotable(), value: M.any() };
+export const ratioPattern = {
```

Comment:
They'll be part of a hardened structure, but it would still be a mistake if anyone modified them before using them. Wouldn't it be helpful to `harden()` them?

Reply:
eek! yes, of course; lack of `harden()` is a pretty bad hazzard here.

Reply:
`harden()` is designed to be used on all newly created objects before they escape the static scope in which they are created:

> `harden()` should be called on all objects that will be transferred across a trust boundary. It's important to `harden()` an object before exposing the object by returning it or passing it to some other function.
 -- https://agoric.com/documentation/guides/js-programming/hardened-js.html#defensive-objects-with-harden

> To enable sound static reasoning, in Jessie all objects made by literal expressions (object literals, array literals, the many forms of function literals) must be tamper-proofed with harden before they can be aliased or escape from their static context of origin.
 -- https://github.com/endojs/Jessie

This supports defensive correctness at a granularity finer than vats -- all the way down to individual objects.

In the interest of programmer convenience, in some of our APIs, the callee hardens for the caller. I think @erights is uneasy in all such cases, since (1) it risks masking bugs / confusion (as discussed in https://agoric.com/documentation/guides/js-programming/far.html#pass-styles-and-harden ) , and (2) some callers might be surprised by having their objects hardened as a consequence of passing them to functions such as `fit()`.